agr_java_core/src/main/java/org/alliancegenome/core/variant/converters/AlleleVariantSequenceConverter.java

```diff
@@ -112,27 +113,24 @@
 
             agrAllele.setSymbol(hgvsNomenclature);
             agrAllele.setSymbolText(hgvsNomenclature);
+            Set<String> synonymSet = new HashSet<>();
+            Set<String> crossReferencesSet = new HashSet<>();
             if (htpConsequences != null) {
                 for (TranscriptLevelConsequence consequence : htpConsequences) {
                     Gene consequenceGene = consequence.getAssociatedGene();
-                    if(geneCache != null && consequenceGene != null) {
-                        Set<String> synonymSet = geneCache.getSynonyms().get(consequenceGene.getPrimaryKey());
-                        Set<String> crossReferencesSet = geneCache.getCrossReferences().get(consequenceGene.getPrimaryKey());
-                        if(synonymSet != null) {
-                            List<String> synonymList = new ArrayList<>(synonymSet);
-                            consequenceGene.setSynonymList(synonymList);
-                        }
-
-                        if(crossReferencesSet != null) {
-                            List<String> crossReferencesList = new ArrayList<>(crossReferencesSet);
-                            consequenceGene.setCrossReferencesList(crossReferencesList);
-                        }
-                        
-                    }
 
                     if(consequenceGene != null) {
                         consequenceGene.setSpecies(species);
                     }
+
+                    if(firstConsequence){
```

Comment:
I created a second first check because I wasn't sure if it would make sense to have this logic under the `if(!transcriptsProcessed.contains(transcriptID))` check

Reply:
Why not just `break;` inside the if statement once the synonym and crossref's are set?

Reply:
Oh never mind the rest of the loop needs to run for each CSQ

Reply:
Yeah, I still feel like there's a better way than what I have.

Reply:
You might want to put the `firstConsequence = false;` inside the if statement so they will be set.

Reply:
So put it within the `if(geneCache != null && consequenceGene != null)` check? Would that be in case that statement evaluates to true on an iteration other than the first, but not on the first iteration?